src/c/device-can.c

```diff
@@ -230,47 +227,64 @@ static bool can_put_handler(void *impl, const devsdk_device_t *device,
 
 	can_driver *driver = (can_driver *)impl;
 	struct can_frame frame;
+	int iret = -1;
+	bool successful_put_request = false;
+	uint8_t i = 0, k = 0; 
 
-	bool successful_get_request = false;
-	uint8_t i = 0; 
+	end_dev_params *end_dev_params_ptr = (end_dev_params *)device->address;
 
 	pthread_mutex_lock(&driver->mutex);
 
-	// Set the iterator to the first array
-	iot_data_array_iter_t iter;
-	iot_data_array_iter (values[0], &iter);
-
-	iot_log_debug(driver->lc, "CAN: PUT on device \n");
-	iot_log_debug(driver->lc, "CAN: nvalues = %d\n", nvalues);
+	if(!end_dev_params_ptr->can_IsOpened)
+	{
+		iret = OpenCan(impl, device);
+		if(iret != 0)
+			return successful_put_request;
```

Comment:
the mutex should be unlocked before the `return` statement

Reply:
Modified the code